src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/OffsetOfTests.cs

```diff
@@ -0,0 +1,364 @@
+        public ushort wYear;
+        [FieldOffset(2)]
+        public ushort wMonth;
+        [FieldOffset(4)]
+        public ushort wDayOfWeek;
+        [FieldOffset(6)]
+        public ushort wDay;
+        [FieldOffset(8)]
+        public ushort wHour;
+        [FieldOffset(10)]
+        public ushort wMinute;
+        [FieldOffset(12)]
+        public ushort wSecond;
+        [FieldOffset(14)]
+        public ushort wMilliseconds;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public class MyPoint
+    {
+        public int x;
+        public int y;
+    }
+
+    public class NoLayoutPoint
+    {
+        public int x;
+        public int y;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    public class NonExistField
+    {
+
+    }
+
+    [StructLayout(LayoutKind.Explicit)]
+    internal struct ExplicitLayoutTest
+    {
+        [FieldOffset(0)]
+        public short m_short1; // 2 bytes
+        [FieldOffset(2)]
+        public short m_short2; // 2 bytes
+
+        [FieldOffset(4)]
+        public byte union1_byte1; // 1 byte
+        [FieldOffset(5)]
+        public byte union1_byte2; // 1 byte
+        [FieldOffset(6)]
+        public short union1_short1; // 2 bytes
+        [FieldOffset(8)]
+        public Int32 union1_int1; // 4 bytes
+        [FieldOffset(12)]
+        public Int32 union1_int2; // 4 bytes
+        [FieldOffset(16)]
+        public double union1_double1; // 8 bytes
+
+        [FieldOffset(4)]
+        public ushort union2_ushort1; // 2 bytes
+        [FieldOffset(6)]
+        public ushort union2_ushort2; // 2 bytes
+        [FieldOffset(8)]
+        public Int32 union3_int1; // 4 bytes
+        [FieldOffset(8)]
+        public decimal union3_decimal1; // 16 bytes
+
+        [FieldOffset(24)]
+        public ushort m_ushort1; // 2 bytes
+                                 // 6 bytes of padding
+
+        [FieldOffset(32)]
+        public decimal m_decimal1; // 16 bytes
+
+        [FieldOffset(48)]
+        public char m_char1; // 1 byte
+                             // 7 bytes of padding
+    }
+
+    internal struct FieldAlignmentTest
+    {
+        public byte m_byte1; // 1 byte
+                             // 1 byte of padding
+
+        public short m_short1; // 2 bytes
+        public short m_short2; // 2 bytes
+                               // 2 bytes of padding
+
+        public Int32 m_int1; // 4 bytes
+        public byte m_byte2; // 1 byte
+                             // 3 bytes of padding
+
+        public Int32 m_int2; // 4 bytes
+                             // 4 bytes of padding (0 bytes on x86/Unix according System V ABI as double 4-byte aligned)
+
+        public double m_double1; // 8 bytes
+        public char m_char1; // 1 byte
+        public char m_char2; // 1 byte
+        public char m_char3; // 1 byte
+                             // 5 bytes of padding (1 byte on x86/Unix according System V ABI as double 4-byte aligned)
+
+        public double m_double2; // 8 bytes
```

Comment:
This padding for double might be incorrect for linux, since The System V ABI for i386 defines 4-byte alignment for 64-bit types.

Reply:
Do you have any suggested change for this? I'm sorry that I'm not familiar with this for linux

Reply:
I would suggest mark TestFieldAlignment() as [PlatformSpecific(TestPlatforms.Windows)].
We can add a testcase for 64 bits types alignment specified for linux later.

Reply:
Scoped the test for windows only now.